packages/lit-html/src/directive-helpers.ts

```diff
@@ -162,17 +163,30 @@ export const insertPart = (
     }
     if (endNode !== refNode || parentChanged) {
       let start: Node | null = part._$startNode;
+      // moveBefore() cannot be called if the old parent and new parent do not
+      // have the same connected shadow-including-root, so we fall back to
```

Comment:
```suggestion
      // have the same shadow-including-root, so we fall back to
```

moveBefore() also cannot be called if they do not share the same _disconnected_ shadow-including-root! The shadow-including-root must be the same, regardless of connection status. The condition below is correct, it's just this documentation that I'd tweak.

Reply:
Thanks @domfarolino!

btw, I've never benchmarked `getRootNode({composed: true})`, nor used it in a hot code path like this. I'm somewhat concerned about the performance impact of calling it twice here if the composed root has to be looked up with an ancestor walk. I'm wondering if I should do other checks first, like `_endNode.parentNode === container`.

And while I'm dreaming about more ergonomic move APIs, `node.hasSameRoot(otherNode, {composed: true})` might be nice if it can do a faster check than two composed root lookups.